models/userModel.js

```diff
@@ -0,0 +1,100 @@
+const _ = require('lodash');
+const jwt = require('jsonwebtoken');
+const crypto = require('crypto-js');
+const config = require('config');
+const Requests = require('utilities/helpers/api/requests');
+const { User } = require('database').models;
+
+const destroyLastSession = async ({ user }) => {
+  if (_.get(user, 'auth.sessions', false) && user.auth.sessions.length > config.limit_sessions) {
+    await User.updateOne({ _id: user._id }, { $pull: { 'auth.sessions': { _id: user.auth.sessions[0]._id } } });
+  }
+};
+
+const destroySession = async ({ userId, session }) => {
+  await User.updateOne({ _id: userId }, { $pull: { 'auth.sessions': { _id: session._id } } });
+};
+
+const findUserBySocial = async ({ id, provider }) => User.findOne({ 'auth.provider': provider, 'auth.id': id });
+
+const findUserById = async (id) => {
+  try {
+    return { user: await User.findOne({ _id: id }).lean() };
+  } catch (error) {
+    return { error };
+  }
+};
+
+const findUserByName = async ({ name }) => User.findOne({ name });
+
+const signUpSocial = async ({
+  userName, alias, provider, avatar, id, session, email,
+}) => {
+  if (avatar) avatar = await Requests.uploadAvatar({ userName, imageUrl: avatar });
```

Comment:
hello) i think you want write image download module)) you want to save images to waivio image server?) you have 2 ways, register amazon sc3 image server(it cost money, you dont have budget), or save users avatars as they are, or the third option, ask Roma if you can use the Vaivio server to save pictures